src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/dagligReise/beregning/DagligReiseBeregningService.kt

```diff
@@ -32,31 +40,59 @@ class DagligReiseBeregningService(
             typeVedtak = typeVedtak,
         )
 
-        val oppfylteVilkår = vilkårService.hentOppfylteDagligReiseVilkår(behandlingId).map { it.mapTilVilkårDagligReise() }
+        val oppfylteVilkår =
+            vilkårService.hentOppfylteDagligReiseVilkår(behandlingId).map { it.mapTilVilkårDagligReise() }
         validerFinnesReiser(oppfylteVilkår)
 
         val oppfylteVilkårGruppertPåType = oppfylteVilkår.filter { it.fakta != null }.groupBy { it.fakta!!.type }
 
         return BeregningsresultatDagligReise(
-            offentligTransport = beregnOffentligTransport(oppfylteVilkårGruppertPåType, vedtaksperioder),
+            offentligTransport = beregnOffentligTransport(oppfylteVilkårGruppertPåType, vedtaksperioder, behandling),
         )
     }
 
     private fun beregnOffentligTransport(
         vilkår: Map<TypeDagligReise, List<VilkårDagligReise>>,
         vedtaksperioder: List<Vedtaksperiode>,
+        behandling: Saksbehandling,
     ): BeregningsresultatOffentligTransport? {
         val vilkårOffentligTransport = vilkår[TypeDagligReise.OFFENTLIG_TRANSPORT] ?: return null
 
-        return offentligTransportBeregningService.beregn(
-            vedtaksperioder = vedtaksperioder,
-            oppfylteVilkår = vilkårOffentligTransport,
-        )
+        val forrigeVedtak = hentForrigeVedtak(behandling)
+
+        val beregnignsresultat =
+            offentligTransportBeregningService.beregn(
+                vedtaksperioder = vedtaksperioder,
+                oppfylteVilkår = vilkårOffentligTransport,
+            )
+
+        if (forrigeVedtak != null) {
+            val tidligsteEndring =
+                utledTidligsteEndringService.utledTidligsteEndringForBeregning(
+                    behandlingId = behandling.id,
+                    vedtaksperioder = vedtaksperioder,
+                )
+
+            offentligTransportBeregningValidering.validerRevurdering(
```

Comment:
Ser helst at vi splitter denne opp i to funksjoner som er navngitt, typ
`validerHarBlittGjortEndringer`
`kastFeilHvisAktivPeriodeEndresTilMånedskort`